emcc

```diff
@@ -1340,14 +1344,42 @@ try:
 
   js_transform_tempfiles = [final]
 
-  if memory_init_file:
+  if memory_init_file or shared.Settings.MEM_INIT_METHOD == 2:
     memfile = target + '.mem'
     shared.try_delete(memfile)
     def repl(m):
       # handle chunking of the memory initializer
       s = m.groups(0)[0]
       if len(s) == 0 and not shared.Settings.EMTERPRETIFY: return m.group(0) # emterpreter must have a mem init file; otherwise, don't emit 0-size ones
-      open(memfile, 'wb').write(''.join(map(lambda x: chr(int(x or '0')), s.split(','))))
+      membytes = [int(x or '0') for x in s.split(',')]
+      if not shared.Settings.EMTERPRETIFY:
```

Comment:
This was mostly because if the memory is all null bytes, then I do a `return ''`, which I considered semantically equivalent to `return m.group(0)`. Come to think of it, if it indeed is, then perhaps I should make the two lines agree. Can I use `return ''` in both cases? Anyway, the `return m.group(0)` is *not* used for `EMTERPRETIFY`, presumably because we need a mem file there, even if it's empty. If it is OK for the emterpreter to have trailing zeros stripped, then I can move that condition to the `if not membytes` case.